### PR TITLE
Specter2: Frame libraries

### PR DESCRIPTION
**Pull request by lynnerang**
Library feature includes:
- auto-detection of library based on DS components in frame
- ability to set multiple libraries from home tab, including custom libraries (user named)
- restricts library selection so Mercado, Mercado MVP, and Art Deco are mutually exclusive
- shows validation error under component if component metadata library conflicts with frame settings
- shows the library in the legend if there are multiple libraries selected on the frame or it conflicts with the frame libraries
- appends the platform and library names to each spec slide title

Screen Shot 2022-06-27 at 3 17 22 PM
Screen Shot 2022-06-27 at 3 19 48 PM
Screen Shot 2022-06-27 at 3 20 15 PM

### CODE REVIEW

File: specter2/src/App.ts [hunk 1]
```diff
@@ -1148,6 +1150,18 @@ export default class App {
           setData(frame, 'platform', primaryPlatform);
         }
 
+        if (!libraries?.length) {
+          // look for any component with a library assigned to default to
+          const frameHasLibrary = value => !!frame.findOne(
```
Comment (gblakeman) [inline specter2/src/App.ts:1155]: Might just be me, but this is pretty hard to follow. You’re setting a boolean, right? And the boolean is `true` in what scenario? Maybe an example in the comment? Or break it out into a check and sub-check?

File: specter2/src/Painter.ts [hunk 2]
```diff
@@ -661,9 +683,15 @@ export default class Painter {
     }
 
     const platform: Platform = getData(designClone, 'platform') || 'web';
+    const libraries: Libraries = getData(designClone, 'libraries') || ['art-deco'];
     const slide: FrameNode = await importInstance(SPEC_KEYS[slideType].specKey);
     slide.name = `${slide.name} - ${this.frame.name}`;
 
+    const slideTitle = slide.findOne(layer => layer.name === 'Title') as TextNode;
+    const platformText = PLATFORM_OPTS.find(opt => opt.value === platform).text;
+    const libraryText = libraries.map(lib => LIBRARY_OPTS.find(({ value }) => value === lib)?.text || lib).join(', ');
```
Comment (gblakeman) [inline specter2/src/Painter.ts:692]: Another one with enough chaining that it’s hard for me to follow from a readability standpoint. Could it be broken up a bit?

**Review by gblakeman — APPROVED** [review]
🥳🎈

Looks good! I flagged a few readability things I struggled with.

File: specter2/src/App.ts (hunk 1 above)
Comment (lynnerang) [reply specter2/src/App.ts]: just updated the comment and parameter name, and added the boolean return type if you want to take another look. it's a function where I can pass in a library (mercado/mvp/art deco) and it returns true if there's a component within the frame with that library assigned, so we can default the overall frame library to that.

File: specter2/src/Painter.ts (hunk 2 above)
Comment (lynnerang) [reply specter2/src/Painter.ts]: yepyep, just extracted the display name getter to a tool and swapped it out

File: specter2/src/App.ts (hunk 1 above)
Comment (gblakeman) [reply specter2/src/App.ts]: Nice, these updates help me from a readability perspective!

Comment (lynnerang) [reply specter2/src/App.ts]: Sweeeet, glad to hear!
